crates/ruff_linter/src/settings/types.rs

```diff
@@ -289,6 +289,81 @@ impl FromStr for PatternPrefixPair {
     }
 }
 
+#[derive(
+    Clone,
+    Copy,
+    Debug,
+    PartialOrd,
+    Ord,
+    PartialEq,
+    Eq,
+    Default,
+    Serialize,
+    Deserialize,
+    CacheKey,
+    EnumIter,
+)]
+#[cfg_attr(feature = "clap", derive(clap::ValueEnum))]
+#[serde(rename_all = "lowercase")]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+pub enum Language {
+    #[default]
+    Python,
+    Pyi,
+    Ipynb,
+}
```

Comment:
This is not a blocker but I don't think `Ipynb` is the correct term here as that's just an extension. Although I'm not sure what language name corresponds here. Pre-commit uses `jupyter` but we moved away from using that previously so I think this might be fine for now.

Reply:
Could `Notebook` be the right word? Maybe it's not specific enough.  Equally, `Pyi` maybe should be `Stub`?

Reply:
I think all of these file types contain Python syntax, although a Notebook might have additional syntax as well, so the language used is Python technically :)

It's fine to use the file extension as a fallback.